src/ImageSharp/Formats/Bmp/BmpEncoderCore.cs

```diff
@@ -276,5 +288,51 @@ private void Write16Bit<TPixel>(Stream stream, Buffer2D<TPixel> pixels)
                 }
             }
         }
+
+        /// <summary>
+        /// Writes an 8 Bit image with a color palette. The color palette has 256 entry's with 4 bytes for each entry.
+        /// </summary>
+        /// <typeparam name="TPixel">The type of the pixel.</typeparam>
+        /// <param name="stream">The <see cref="Stream"/> to write to.</param>
+        /// <param name="image"> The <see cref="ImageFrame{TPixel}"/> containing pixel data.</param>
+        private void Write8Bit<TPixel>(Stream stream, ImageFrame<TPixel> image)
+            where TPixel : struct, IPixel<TPixel>
+        {
+#if NETCOREAPP2_1
+            Span<byte> colorPalette = stackalloc byte[ColorPaletteSize8Bit];
+#else
+            byte[] colorPalette = new byte[ColorPaletteSize8Bit];
+#endif
+
+            var quantizer = new OctreeQuantizer(dither: true, maxColors: 256);
+            QuantizedFrame<TPixel> quantized = quantizer.CreateFrameQuantizer<TPixel>(this.configuration).QuantizeFrame(image);
+
+            int idx = 0;
+            var color = default(Rgba32);
+            foreach (TPixel quantizedColor in quantized.Palette)
+            {
+                quantizedColor.ToRgba32(ref color);
```

Comment:
Yeah, in Octree we're working in the `Rgba32` space then converting to/from so I see where you are coming from. However, by converting before quantization we end up allocating a new buffer to house the image. That's probably a lot more costly.

Reply:
Totally forgot that it will lead to another allocation, so let's keep it for now.

(An idea is starting to formulate in my mind, but of course it would break `IQuantizer` API again 😄.)